pkg/chain/evmregistry.go

```diff
@@ -168,6 +170,14 @@ func (r *evmRegistryv2_0) CheckUpkeep(ctx context.Context, key types.UpkeepKey)
 	result.CheckBlockHash = ret0.Result.CheckBlockhash
 	result.PerformData = ret0.Result.PerformData
 
+	// reset key to check block number returned from contract
+	// this is the result that will be used in the cache both to reduce repeated
+	// check attempts and provide a lookup for in-flight performs
+	// the reason for this is in the case that the block the upkeep is checked
+	// at is different than what the contract returned. the contract value
+	// takes priority.
+	result.Key = BlockAndIdToKey(big.NewInt(int64(result.CheckBlockNumber+1)), upkeepId)
```

Comment:
not sure why you need blockNumber + 1?

Reply:
This is the part we should talk about over a call. There were some changes to the registry and I'm trying to ensure keys line up everywhere. Rebuilding the key may not be necessary, but let's talk through it to be sure.